### PR TITLE
simplify actions

### PR DESCRIPTION
**Pull request by ildyria**
This PR just aim to remove the intrinsic complexity of the yaml cache etc and hopefully fix the random pipelines errors.

### CODE REVIEW

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -60,117 +60,90 @@ jobs:
...
+        with:
+          composer-options: "${{ matrix.mode }}"
+
       # begin of DEV
-      - name: Install Composer dependencies (dev)
-        if: ${{ matrix.mode == 'dev' }}
-        run: composer install --prefer-dist --no-interaction
+      - name: Install Composer dependencies
+        if: ${{ matrix.mode == '' }}
+        uses: "ramsey/composer-install@v2"
```
Comment (d7415) [inline .github/workflows/Build.yml]: Isn't this covered by the section above now?

Comment (ildyria) [reply .github/workflows/Build.yml]: isn't this already fixed ? :'D

Comment (ildyria) [reply .github/workflows/Build.yml]: https://github.com/LycheeOrg/Lychee/pull/1335/commits/56c1c8149a51ca87ca50c842df580ec7d362e1bd

File: .github/workflows/Build.yml [hunk 2]
```diff
@@ -60,117 +60,86 @@ jobs:
...
 
       - name: setup MySQL Database with v3
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
         run: |
           mysql -uroot -proot homestead_test < .github/workflows/v3.sql
 
       - name: Generate secure key & Migrate & go backward
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
```
Comment (d7415) [inline .github/workflows/Build.yml:142]: If we removed the separation of dev and dist and integrated them a little we could use the artisan steps above for this to remove some more duplication. That's not a deal breaker though.

**Review by d7415 — APPROVED** [review]
One possible improvement, but it would change the structure so you might prefer not to.

**Comment by kamil4** [comment]
I feel unqualified to review this. I lack any understanding of Github actions...

**Review by qwerty287 — APPROVED** [review]
I don't use GitHub Actions as well, but it looks good so far.

**Pull request merged**